### PR TITLE
Set timeout for long tests

### PR DESCRIPTION
**Pull request by karczex**
not tested ;)

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemkv/803)

### CODE REVIEW

File: utils/docker/run-build.sh [hunk 1]
```diff
@@ -17,6 +17,12 @@ BUILD_JSON_CONFIG=${BUILD_JSON_CONFIG:-ON}
 CHECK_CPP_STYLE=${CHECK_CPP_STYLE:-ON}
 TESTS_LONG=${TESTS_LONG:-OFF}
 
+#set arbitrary test timeouts
+if [ "TESTS_LONG" == "OFF"]; then
```
Comment (lukaszstolarczuk) [inline utils/docker/run-build.sh:21]: "${TESTS_LONG}"

Comment (lukaszstolarczuk) [inline utils/docker/run-build.sh:21]: and you can add this to run-tests-building.sh, I guess

**Comment by karczex** [comment]
___
*[utils/docker/run-build.sh, line 21 at r1](https://reviewable.io/reviews/pmem/pmemkv/803#-MIZxsfN7u02TfF_nfjw-r1-21:-MI_7h9K04yKPt-VgJw0:b6g2um8) ([raw file](https://github.com/pmem/pmemkv/blob/0be0a6cd36bb8e6325587adb4ff0d43834778738/utils/docker/run-build.sh#L21)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

and you can add this to run-tests-building.sh, I guess
</blockquote></details>

It's not needed in run-tests-building.sh

**Comment by karczex** [comment]
___
*[utils/docker/run-build.sh, line 21 at r1](https://reviewable.io/reviews/pmem/pmemkv/803#-MIZxsfN7u02TfF_nfjw-r1-21:-MI_7q165AIhICHS4UZx:b-896fix) ([raw file](https://github.com/pmem/pmemkv/blob/0be0a6cd36bb8e6325587adb4ff0d43834778738/utils/docker/run-build.sh#L21)):*
<details><summary><i>Previously, karczex wrote…</i></summary><blockquote>

It's not needed in run-tests-building.sh
</blockquote></details>

Done.

**Review by lukaszstolarczuk — COMMENTED** [review]
> Reviewed 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv/803)* status: all files reviewed, 1 unresolved discussion (waiting on @karczex)

___
*[utils/docker/run-build.sh, line 21 at r1](https://reviewable.io/reviews/pmem/pmemkv/803#-MIZxsfN7u02TfF_nfjw-r1-21:-MIc8V1A2z4Y4tWj8tm4:b-3ct6jw) ([raw file](https://github.com/pmem/pmemkv/blob/0be0a6cd36bb8e6325587adb4ff0d43834778738/utils/docker/run-build.sh#L21)):*
<details><summary><i>Previously, karczex wrote…</i></summary><blockquote>

Done.
</blockquote></details>

we don't set timeout there, but I guess we should/could ;)

**Review by igchor — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv/803)* status: all files reviewed, 1 unresolved discussion (waiting on @karczex)

___
*[utils/docker/run-build.sh, line 22 at r2](https://reviewable.io/reviews/pmem/pmemkv/803#-MIcFhhI3bPcyz34615S:-MIcFhhJ6TlVDp15mGU9:b-ribe2u) ([raw file](https://github.com/pmem/pmemkv/blob/ea5a84b1fd087fde1f6262579220473eda162142/utils/docker/run-build.sh#L22)):*
> ```Shell
> 
> #set arbitrary test timeouts
> if [ "${TESTS_LONG}" == "OFF"]; then
> ```

Hm, maybe we could just expose TEST_TIMEOUT instead of hardocoding it? (with default value equal to 590). If, in future, some tests will require more than 3600s we can just change it in jenkins params instead of here.

**Review by karczex — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv/803)* status: all files reviewed, 1 unresolved discussion (waiting on @lukaszstolarczuk)

___
*[utils/docker/run-build.sh, line 21 at r1](https://reviewable.io/reviews/pmem/pmemkv/803#-MIZxsfN7u02TfF_nfjw-r1-21:-MIcOhUsElBBNogezPoL:bcsp67j) ([raw file](https://github.com/pmem/pmemkv/blob/0be0a6cd36bb8e6325587adb4ff0d43834778738/utils/docker/run-build.sh#L21)):*
<details><summary><i>Previously, lukaszstolarczuk (Łukasz Stolarczuk) wrote…</i></summary><blockquote>

we don't set timeout there, but I guess we should/could ;)
</blockquote></details>

nope. (see next comment)
___
*[utils/docker/run-build.sh, line 22 at r2](https://reviewable.io/reviews/pmem/pmemkv/803#-MIcFhhI3bPcyz34615S:-MIcNaI47KPDZQTtHdEa:b-kvdx4a) ([raw file](https://github.com/pmem/pmemkv/blob/ea5a84b1fd087fde1f6262579220473eda162142/utils/docker/run-build.sh#L22)):*
<details><summary><i>Previously, igchor (Igor Chorążewicz) wrote…</i></summary><blockquote>

Hm, maybe we could just expose TEST_TIMEOUT instead of hardocoding it? (with default value equal to 590). If, in future, some tests will require more than 3600s we can just change it in jenkins params instead of here.
</blockquote></details>

This is quick workaround. IMHO Global timeout parameter is wrong in general; we shouldn't expose it at all. Instead of this we should set timeout per test (or tests group) in cmake.

**Review by igchor — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pmem/pmemkv/803)* status: all files reviewed, 1 unresolved discussion (waiting on @lukaszstolarczuk)

___
*[utils/docker/run-build.sh, line 22 at r2](https://reviewable.io/reviews/pmem/pmemkv/803#-MIcFhhI3bPcyz34615S:-MIcbUe-5WV8JBjg79UM:bjvhitk) ([raw file](https://github.com/pmem/pmemkv/blob/ea5a84b1fd087fde1f6262579220473eda162142/utils/docker/run-build.sh#L22)):*
<details><summary><i>Previously, karczex wrote…</i></summary><blockquote>

This is quick workaround. IMHO Global timeout parameter is wrong in general; we shouldn't expose it at all. Instead of this we should set timeout per test (or tests group) in cmake. 
</blockquote></details>

I'm not sure if it should be set inside CMake. Test execution time will depend on your platform, whether you run the tests on pmem, etc.

**Comment by lukaszstolarczuk** [comment]
as agreed with karczex, this changes are now included into bigger change: #890

**Pull request closed without merging**